### ISSUE TITLE
Track 3: test Kaon as an optimizer-only Muon ablation

### ISSUE THREAD

**Issue by tokenbender** [KellerJordan/modded-nanogpt#296]
## Summary

Test **Kaon** from arXiv `2605.11181`, [Muon is Not That Special: Random or Inverted Spectra Work Just as Well](https://arxiv.org/abs/2605.11181), as a Track 3 optimizer-only ablation.

The important point is that Kaon does **not** require architecture, dataset, batch size, or training-loop changes. It is just a replacement for the Muon spectral/orthogonalization map inside the matrix-parameter optimizer.

## Why this is worth testing

The paper argues that Muon's gains may not depend on exact adherence to a special non-Euclidean geometry. Their counterexample is Kaon: an intentionally odd optimizer that replaces the gradient singular values with a chaotic/random spectrum and still closely matches Muon in their NanoGPT/WikiText-2 experiments.

The relevant claim to test here is narrow:

- Keep Track 3 fixed.
- Keep the current model and data path fixed.
- Replace the Muon direction transform with Kaon's chaotic spectral map.
- Retune LR/WD and compare step count to the existing Track 3 Muon/Contra/Soft-Muon baselines.

## Exact Kaon update from the paper

Paper Algorithm 2 gives Kaon as:

```text
X0 = G / ||G||F
for t = 1..T:
    At = Xt-1 Xt-1^T
    Bt = (I - At)^2
    Xt = 4.1 * Bt Xt-1
return XT / 1.175
```

In PyTorch-like form:

```python
def kaon_via_chaos(G, steps=5):
    X = G.bfloat16()
    transposed = G.size(-2) > G.size(-1)
    if transposed:
        X = X.mT

    X = X / (X.norm(dim=(-2, -1), keepdim=True) + 1e-7)
    I = torch.eye(X.size(-2), device=X.device, dtype=X.dtype)
    for _ in range(steps):
        A = X @ X.mT
        B = I - A
        B = B @ B
        X = 4.1 * (B @ X)

    X = X / 1.175
    if transposed:
        X = X.mT
    return X
```

Notes from the paper:

- Kaon uses the recurrence `x_{t+1} = lambda * x_t * (1 - x_t^2)^2` lifted to matrix iterations.
- The paper sets `lambda = 4.1`.
- The appendix describes the resulting chaotic stationary support as roughly `(0, 1.175)`, hence the final `/ 1.175` scale.
- Their training detail says Kaon used the same Muon-style momentum `0.95` and `5` matrix iterations.
- Their selected NanoGPT/WikiText-2 tuning table used `matrix_lr = 2e-2`, `base_lr = 8e-3` for Kaon, and their dense validation plot used Kaon at a scaled setting `rho = 3`. These values should not be copied blindly into Track 3, but they confirm LR/WD retuning is expected.

## Where this maps into Track 3 code

The simple baseline currently performs Muon here:

```python
def zeropower_via_newtonschulz5(G: Tensor) -> Tensor:
    ...

@torch.compile
def muon_update(grad, momentum, mu=0.95, nesterov=True):
    momentum.lerp_(grad, 1 - mu)
    update = grad.lerp_(momentum, mu) if nesterov else momentum
    update = zeropower_via_newtonschulz5(update)
    update *= max(1, grad.size(-2) / grad.size(-1))**0.5
    return update
```

So the smallest test is to add `kaon_via_chaos` next to `zeropower_via_newtonschulz5` and replace only:

```python
update = zeropower_via_newtonschulz5(update)
```

with:

```python
update = kaon_via_chaos(update, steps=5)
```

Everything else should initially remain fixed:

- momentum update
- shape scaling
- weight decay application
- distributed parameter sharding/all-gather
- aux Adam parameter groups
- Track 3 architecture/data/batch/training rules

## Suggested first experiments

1. **Pure Kaon baseline**
   - Start from `records/track_3_optimization/train_gpt_simple.py`.
   - Swap Muon NS/polar map for Kaon.
   - Sweep Muon matrix LR and WD around current Track 3 ranges.

2. **Kaon in current record-style optimizer**
   - Start from the Contra/Soft-Muon repro path.
   - Replace the late Soft-Muon target with Kaon, while keeping norm matching.
   - This isolates whether chaotic spectrum reshaping works as a drop-in late-training direction change.

3. **Blend ablation**
   - Compare `ns_update`, `contra_update`, `soft_update`, and `kaon_update` under the same norm normalization.
   - Try late blend schedules first, because existing Track 3 results already suggest optimizer direction interpolation can matter late.

## What not to change initially

- Do not change architecture.
- Do not change dataset or batch size.
- Do not add second forward/backward passes.
- Do not import third-party optimizer libraries.
- Do not implement Freon in this issue. This is only for Kaon.

## Acceptance signal

A useful result does not need to beat the current global record immediately. It would still be valuable if it establishes one of:

- Kaon is competitive with tuned Muon in Track 3.
- Kaon is not competitive after a fair LR/WD sweep.
- Kaon helps only as a late blend or only on specific parameter groups.
- Kaon fails due to numerical or scale issues under Track 3's current bf16/distributed setup.

The cleanest result would report mean final validation loss over non-cherry-picked seeds at a fixed step count, following the existing Track 3 statistical rule.

## References

- Paper: https://arxiv.org/abs/2605.11181
- PDF: https://arxiv.org/pdf/2605.11181
- Original discussion pointer: https://x.com/tokenbender/status/2054544369118855491
